### PR TITLE
enable incrementalbuild for managedreference plugins except splitmemberlevel

### PR DESCRIPTION
**Pull request by ansyral**
1.  some methods in IncrementalBuildContext aren't  thread-safe and when managedreferenceprocessor enables incremental, two processors run in parallel might have some problems, fix it.
2.  enable plugins incremental except for splitmemberlevel.

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs [hunk 1]
```diff
@@ -371,7 +375,10 @@ public ProcessorInfo CreateProcessorInfo(IDocumentProcessor processor)
                     IncrementalContextHash = ((ISupportIncrementalBuildStep)step).GetIncrementalContextHash(),
                 });
             }
-            CurrentBuildVersionInfo.Processors.Add(cpi);
+            lock (_sync)
+            {
+                CurrentBuildVersionInfo.Processors.Add(cpi);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs]: >CurrentBuildVersionInfo [](start = 16, length = 23)

why should lock this and _modelLoadInfo>

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs]: processors are run in parallel

**Comment by vwxyzh** [comment]
:shipit:

**Pull request merged**